core/model/configuration.py

```diff
@@ -1154,7 +1159,11 @@ def __get__(
         return setting_value
 
     def __set__(
-        self, owner_instance: HasConfigurationSettings | None, value: Any
+        self,
+        owner_instance: HasConfigurationSettings
+        | IgnoredIdentifierConfiguration
```

Comment:
This is an interesting Union type, is IgnoreIdentifierConfiguration special, or should it also inherit from HasConfigurationSettings?

Reply:
IgnoreIdentifierConfiguration is not inheriting from HasConfigurationSettings, instead it [inherits ConfigurationTrait](https://github.com/ThePalaceProject/circulation/blob/main/core/configuration/ignored_identifier.py#L15) (although that class doesn't do anything) so mypy complains here. 

```
core/configuration/ignored_identifier.py:75: error: Argument 1 to "__set__" of "ConfigurationMetadata" has incompatible type "IgnoredIdentifierConfiguration"; expected "Optional[HasConfigurationSettings]"
```

Reply:
Yea it looks like it should functionally be a `HasConfigurationSettings` but has the methods implemented with different names. This sounds like a needed change, but not in this ticket. We can keep it as is, for now.